.github/workflows/e2e.yml

```diff
@@ -0,0 +1,113 @@
+name: E2E NTLM Tests
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  ntlm-e2e:
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.24.x, 1.25.x]
+        venv: [windows-2022, windows-2025]
+    name: E2E NTLM Tests (Windows Server)
+    runs-on: ${{ matrix.venv }}
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 # v6.0.0
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Pre-setup IIS Configuration (Windows Server)
+        shell: pwsh
+        run: |
+          Write-Host "Pre-configuring IIS for Windows Server environment..." -ForegroundColor Green
+
+          # Unlock authentication sections that are often locked on Windows Server
+          Write-Host "Unlocking IIS authentication sections..."
+          try {
+            & $env:windir\system32\inetsrv\appcmd.exe unlock config -section:system.webServer/security/authentication/windowsAuthentication
+            & $env:windir\system32\inetsrv\appcmd.exe unlock config -section:system.webServer/security/authentication/anonymousAuthentication
+            Write-Host "Successfully unlocked authentication sections" -ForegroundColor Green
+          } catch {
+            Write-Host "Could not unlock some sections (may already be unlocked): $($_.Exception.Message)" -ForegroundColor Yellow
+          }
+
+      - name: Setup NTLM Test Environment
+        shell: pwsh
+        run: |
+          Write-Host "Setting up NTLM test environment using setup script..." -ForegroundColor Green
+
+          # Run the setup script
+          .\e2e\setup-local-e2e.ps1
+
+          # Set additional environment variables for GitHub Actions
+          Write-Output "NTLM_TEST_USER=ntlmtestuser" >> $env:GITHUB_ENV
+          Write-Output "NTLM_TEST_PASSWORD=TestPass123!" >> $env:GITHUB_ENV
```

Comment:
Better generate a random string each time, don't like hardcoding passwords in CI environments, even if they are fake.